libs/go-libs/httpserver/serverport.go

```diff
@@ -90,15 +90,50 @@ func StartServer(ctx context.Context, bind string, handler http.Handler, options
 	}, nil
 }
 
-func NewHook(addr string, handler http.Handler, options ...func(server *http.Server)) fx.Hook {
+type server struct {
+	listener       net.Listener
+	httpServerOpts []func(server *http.Server)
+}
+
+type serverOpts func(server *server)
+
+func WithListener(listener net.Listener) serverOpts {
+	return func(server *server) {
+		server.listener = listener
+	}
+}
+
+func WithAddress(addr string) serverOpts {
+	return func(server *server) {
+		l, err := net.Listen("tcp", addr)
```

Comment:
Put this inside StartServer function

Reply:
In wich case ? In case the listener is not provided ?